views/intercept/password-recovery-block.vm

```diff
@@ -0,0 +1,107 @@
+##
+## Velocity Template for Display 2FARequiredSoonNotification view-state
+##
+## Velocity context will contain the following properties
+## flowExecutionUrl - the form action location
+## flowRequestContext - the Spring Web Flow RequestContext
+## flowExecutionKey - the SWF execution key (this is built into the flowExecutionUrl)
+## profileRequestContext - root of context tree
+## authenticationContext - context with authentication request information
+## authenticationErrorContext - context with login error state
+## authenticationWarningContext - context with login warning state
+## ldapResponseContext - context with LDAP state (if using native LDAP)
+## rpUIContext - the context with SP UI information from the metadata
+## extendedAuthenticationFlows - collection of "extended" AuthenticationFlowDescriptor objects
+## passwordPrincipals - contents of the shibboleth.authn.Password.PrincipalOverride bean
+## encoder - HTMLEncoder class
+## request - HttpServletRequest
+## response - HttpServletResponse
+## environment - Spring Environment object for property resolution
+## custom - arbitrary object injected by deployer
+##
+#set ($rpContext = $profileRequestContext.getSubcontext('net.shibboleth.profile.context.RelyingPartyContext'))
+#set ($username = $authenticationContext.getSubcontext('net.shibboleth.idp.authn.context.UsernamePasswordContext', true).getUsername())
+#set ($passwordEnabled = false)
+#if (!$passwordPrincipals or $passwordPrincipals.isEmpty() or $authenticationContext.isAcceptable($passwordPrincipals))
+  #set ($passwordEnabled = true)
+#end
+##
+<!DOCTYPE html>
+<html>
+  	<head>
+    	<meta charset="utf-8">
+<link rel="stylesheet" type="text/css" href="/css/opensans.css" media="screen, handheld" />
+<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
+<title>UW NetID sign-in</title>
+  <meta name="description" content="">
+  <meta name="author" content="">
+<meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta http-equiv="cleartype" content="on">
+<link rel="stylesheet" type="text/css" href="/css/weblogin-global.css" media="screen, handheld" />
+  <link rel="stylesheet" type="text/css" href="/css/weblogin-enhanced.css" media="screen  and (min-width: 40.5em)" />
+  <!--[if (lt IE 9)&(!IEMobile)]>
+  <link rel="stylesheet" type="text/css" href="/css/weblogin-enhanced.css" />
+  <![endif]-->
+
+
+  	</head>
+
+
+<body>
+
+  <h1 class="visuallyhidden">UW NetID sign-in</h1>
+
+  <div id="container">
+    <div id="main" role="main">
+
+      <h2 class="visuallyhidden" aria-flowto="weblogin_warning">Login</h2>
+      <div class="form">
+          <div><img src="/img/weblogin.png" height="57" width="198" alt="" aria-hidden="true"></div>
+
+          <form name="idprecoveryblockintercept" id="idprecoveryblockinterceptdiv" method="POST" action="$flowExecutionUrl" autocomplete="off">
```

Comment:
Just recording that we're not sure if we need a <form> element here; it works, but seems superfluous.

Reply:
I'm going to leave it for now, we can revisit sometime later when we review all our Shibboleth views, decide which to keep, and fix the rest.